cpp/include/cudf/utilities/traits.hpp

```diff
@@ -261,6 +261,30 @@ constexpr inline bool is_signed_iterator()
   return std::is_signed_v<typename std::iterator_traits<Iterator>::value_type>;
 }
 
+/**
+ * @brief Indicates whether the type `T` is an integral type.
```

Comment:
Should we note that `int128` is _not_ considered integral by this definition? (Either way is fine but that subtlety between `std::` and `cuda::std::` is not always obvious.)

Reply:
I think this one should be `cuda::std::is_integral<T>`. I just need some time for the world to recompile to test if this is ok. If not, I'll add the comment.